### PR TITLE
Fix widget.properties.data being optional

### PR DESCRIPTION
**Pull request by samshara**

### CODE REVIEW

**Review by tnagorra — APPROVED** [review]

File: src/utils/framework.ts [hunk 1]
```diff
@@ -182,10 +183,10 @@ export const createReportStructure = (
         }
 
         if (widget.widgetId === 'matrix2dWidget' && reportStructureVariant === DIMENSION_FIRST) {
-            if (!widget.properties) {
+            const { properties: { data } } = widget as Matrix2dWidgetElement;
```
Comment (AdityaKhatri) [inline src/utils/framework.ts]: Don't we need to check whether or not properties is present?

Comment (samshara) [reply src/utils/framework.ts]: I believe the properties object will always be there.

**Review by tnagorra — APPROVED** [review]

**Review by frozenhelium — APPROVED** [review]

**Review by subinasr — APPROVED** [review]

File: src/utils/framework.ts (hunk 1 above)
Comment (tnagorra) [reply src/utils/framework.ts:188]: Let's make some minor change before we merge this
```suggestion
            const { properties: { data } = {} } = widget as Matrix2dWidgetElement;
```

Comment (tnagorra) [reply src/utils/framework.ts]: Let's add a check for properties

File: src/utils/framework.ts [hunk 2]
```diff
@@ -616,14 +623,16 @@ export function getTextWidgetsFromFramework(framework: FrameworkFields) {
         .filter(w => w.widgetId === 'conditionalWidget')
         .map((conditional) => {
             const {
+                properties: {
+                    data,
+                } = {},
```
Comment (tnagorra) [inline src/utils/framework.ts]: Why not access directly? properties?.data

File: src/utils/framework.ts [hunk 3]
```diff
@@ -665,13 +674,16 @@ export function getContextualWidgetsFromFramework(framework: FrameworkFields) {
...
```
Comment (tnagorra) [inline src/utils/framework.ts]: Why not access directly? properties?.data

File: src/utils/framework.ts [hunk 4]
```diff
@@ -578,13 +584,14 @@ export function getScaleWidgetsData(framework: FrameworkFields, entry: Entry) {
...
-                                data: {
-                                    scaleUnits,
-                                },
+                                data: widgetData,
                             },
                         } = widget as WidgetElement<ScaleWidget>;
 
-                        return scaleUnits.find(v => v.key === attributeData.data.value);
+                        if (widgetData) {
+                            const { scaleUnits } = widgetData;
```
Comment (tnagorra) [inline src/utils/framework.ts]: Are we sure scaleUnits is there?

File: src/utils/framework.ts [hunk 5]
```diff
@@ -558,11 +559,16 @@ export function getScaleWidgetsData(framework: FrameworkFields, entry: Entry) {
...
-
+                properties: {
+                    data,
+                },
+                id,
+            } = conditional as WidgetElement<ConditionalWidget>;
+            if (!data) {
+                return [];
+            }
+            const { widgets: widgetsInsideConditional = [] } = data;
```
Comment (tnagorra) [inline src/utils/framework.ts]:
```suggestion
            const widgetsInsideConditional = properties?.data?.widgets ?? [];
```

File: src/utils/framework.ts [hunk 6]
```diff
@@ -616,14 +623,14 @@ export function getTextWidgetsFromFramework(framework: FrameworkFields) {
...
-
-            const {
-                widgets: widgetsInsideConditional = [],
-            } = ((conditional as WidgetElement<ConditionalWidget>).properties.data || {});
-
+            } = conditional as WidgetElement<ConditionalWidget>;
+            if (!properties?.data) {
+                return [];
+            }
+            const { widgets: widgetsInsideConditional = [] } = properties.data;
```
Comment (tnagorra) [inline src/utils/framework.ts]:
```suggestion
            const widgetsInsideConditional = properties?.data?.widgets ?? [];
```

File: src/utils/framework.ts [hunk 7]
```diff
@@ -665,13 +672,14 @@ export function getContextualWidgetsFromFramework(framework: FrameworkFields) {
...
-            } = conditional;
-            const {
-                widgets: widgetsInsideConditional = [],
-            } = ((conditional as WidgetElement<ConditionalWidget>).properties.data || {});
-
+            } = conditional as WidgetElement<ConditionalWidget>;
+            if (!properties?.data) {
+                return [];
+            }
+            const { widgets: widgetsInsideConditional = [] } = properties.data;
```
Comment (tnagorra) [inline src/utils/framework.ts]:
```suggestion
            const widgetsInsideConditional = properties?.data?.widgets ?? [];
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by tnagorra — APPROVED** [review]

**Pull request closed without merging**